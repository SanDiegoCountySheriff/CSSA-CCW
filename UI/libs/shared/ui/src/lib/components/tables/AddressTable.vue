<template>
  <v-simple-table class="my-3">
    <template #default>
      <thead>
        <tr>
          <th>{{ $t(' Address line 1') }}</th>
          <th>{{ $t('Address line 2') }}</th>
          <th>{{ $t('City') }}</th>
          <th>{{ $t('State') }}</th>
          <th>{{ $t('County') }}</th>
          <th>{{ $t('Zip') }}</th>
          <th>{{ $t('Country') }}</th>
        </tr>
      </thead>
      <tbody>
        <tr
          v-for="(address, index) in addresses"
          :key="index"
        >
          <td>
            {{ address.addressLine1 }}
          </td>
          <td>
            {{ address.addressLine2 }}
          </td>
          <td>
            {{ address.city }}
          </td>
          <td>
            {{ address.state }}
          </td>
          <td>
            {{ address.county }}
          </td>
          <td>
            {{ address.zip }}
          </td>
          <td>
            {{ address.country }}
          </td>
        </tr>
      </tbody>
    </template>
  </v-simple-table>
</template>

<script lang="ts">
import { defineComponent } from 'vue';
import { AddressInfo } from '@shared-ui/types/defualtTypes';

export default defineComponent({
  name: 'AddressTable',
  props: {
    addresses: {
      type: Array<AddressInfo>,
      default: () => [],
    },
  },
});
</script>
