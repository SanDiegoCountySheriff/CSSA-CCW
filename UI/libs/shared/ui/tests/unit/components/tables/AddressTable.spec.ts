/* eslint-disable @typescript-eslint/ban-ts-comment */
import AddressTable from '@shared-ui/components/tables/AddressTable.vue'
import Vuetify from 'vuetify'
import { createLocalVue, mount } from '@vue/test-utils'

const localVue = createLocalVue()

describe('AddressTable', () => {
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  it('should match the snapshot', () => {
    //@ts-ignore
    const wrapper = mount(AddressTable, {
      localVue,
      vuetify,
      mocks: {
        $t: text => {
          text
        },
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Should render the address table in the correct order', () => {
    //@ts-ignore
    const wrapper = mount(AddressTable, {
      localVue,
      vuetify,
      mocks: {
        $t: text => text,
      },
      propsData: {
        addresses: [
          {
            addressLine1: 'line1',
            addressLine2: 'line2',
            city: 'city',
          },
        ],
      },
    })

    expect(wrapper.findAll('td').at(0).element.textContent).toContain('line1')
    expect(wrapper.findAll('td').at(1).element.textContent).toContain('line2')
    expect(wrapper.findAll('td').at(2).element.textContent).toContain('city')
  })
})
