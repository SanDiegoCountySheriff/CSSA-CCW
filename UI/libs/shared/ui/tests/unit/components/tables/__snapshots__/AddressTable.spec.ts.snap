// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressTable should match the snapshot 1`] = `
<div class="container container--fluid">
  <div class="v-data-table my-3 theme--light">
    <div class="v-data-table__wrapper">
      <table>
        <thead>
          <tr>
            <th></th>
            <th></th>
            <th></th>
            <th></th>
            <th></th>
            <th></th>
            <th></th>
            <th></th>
          </tr>
        </thead>
        <tbody></tbody>
      </table>
    </div>
  </div>
</div>
`;
